src/Aspire.Dashboard/Model/ResourceViewModel.cs

```diff
@@ -81,9 +81,9 @@ internal bool MatchesFilter(string filter)
         return null;
     }
 
-    public bool IsResourceHidden()
+    public bool IsResourceHidden(bool showHiddenResources)
     {
-        return IsHidden || KnownState is KnownResourceState.Hidden;
+        return (IsHidden || KnownState is KnownResourceState.Hidden) && !showHiddenResources;
```

Comment:
imo this is slightly more understandable
```suggestion
        if (showHiddenResources)
        {
            return false;
        }
        return IsHidden || KnownState is KnownResourceState.Hidden;
```